src/main/java/uk/gov/hmcts/reform/pdf/service/ResponseExceptionHandler.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.reform.pdf.service;
+
+import feign.FeignException;
+import org.apache.commons.lang.NotImplementedException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+import uk.gov.hmcts.reform.pdf.service.exception.AuthException;
+
+import static org.springframework.http.ResponseEntity.status;
+
+@ControllerAdvice
+public class ResponseExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(NotImplementedException.class)
+    protected ResponseEntity handleNotImplemented(NotImplementedException exc, WebRequest req) {
```

Comment:
Will the exceptions handled by this class be logged somewhere or is this the right place to do it?

Reply:
There can be additional logging in ExceptionHandler but it's purpose is to marshal desired response in case of Exception. We do have request interceptor coming in from `java-logging-spring` library which logs desired request accomplishment in any case